### PR TITLE
feat(snap): add content interface for device configuration

### PR DESCRIPTION
**Pull request by siggiskulason**
The `device-config` content interface allows another snap to seed this device
snap with initial configuration and device profiles. To use, create a new
snap with the following in snapcraft.yaml:

```
slots:
  device-config:
    interface: content
    read: $SNAP_DATA/device-profiles
```

where device-profiles is the name of a directory with the new configuration 
for the device snap. You can then connect the plug in the device snap to 
the slot in your snap, which will replace the configuration in the device snap.
Do this with:

```
sudo snap connect edgex-device-rest:device-config your-snap:device-config
```

Signed-off-by: Siggi Skulason <siggi.skulason@canonical.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-rest-go/blob/master/.github/CONTRIBUTING.md

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## New Imports

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by tonyespy — CHANGES_REQUESTED** [review]
Can you please add the documentation that you included in your PR description to the snap/README file?

Also curious why the device service implements a slot vs. a plug? As typically a snap providing content implements the slot. We did this for the vault token interface because it allowed the consuming snap to name the directories that get created under `/secrets`, and maybe the same applied to the app-service-config profiles content interface?

**Comment by siggiskulason** [comment]
Thanks for the comments, @tonyespy ,

> Can you please add the documentation that you included in your PR description to the snap/README file?
Will do!

> Also curious why the device service implements a slot vs. a plug? As typically a snap providing content implements the slot. We did this for the vault token interface because it allowed the consuming snap to name the directories that get created under `/secrets`, and maybe the same applied to the app-service-config profiles content interface?

Only reason was that it was what we did when adding this earlier to the LLRP snap - see [this commit](https://github.com/canonical/edgex-device-rfid-llrp-go/pull/1/commits/6dee5eea0e8145c161e907bb2b476c3579397961). I think that may have been because it was what we did for the ASC interface - which, as you said, was to handle the requirements of setting up secrets for Vault. I'll change this around and make it into a plug - and also modify the LLRP implementation in a separate PR.

File: snap/README.md [hunk 1]
```diff
@@ -31,3 +31,23 @@ To apply the settings, the service should then be restarted as follows:
...
+
+The `device-config` content interface allows another snap to seed this device
+snap with initial configuration and device profiles. To use, create a new
+snap with the following in snapcraft.yaml:
+
+```
+slots:
+  device-config:
+    interface: content
+    read: $SNAP_DATA/device-profiles
```
Comment (tonyespy) [inline snap/README.md:43]: The only thing I'd change is the name of the `read` directory, as device-profiles is a bit of an overloaded term. I'd suggest just using "config" here.

File: snap/README.md [hunk 2]
```diff
@@ -31,3 +31,23 @@ To apply the settings, the service should then be restarted as follows:
 ```bash
 $ sudo snap restart edgex-device-rest.device-rest-go
 ```
+
+The `device-config` content interface allows another snap to seed this device
+snap with initial configuration and device profiles. To use, create a new
```
Comment (tonyespy) [inline snap/README.md:36]: Maybe add a little bit more here to explain that the other snap can provide both a configuration file and one or more device profiles. It's not totally obvious that's possible...

**Review by tonyespy — CHANGES_REQUESTED** [review]
LGTM, however some minor changes requested for snap/README.

File: snap/README.md (hunk 1 above)
Comment (siggiskulason) [reply snap/README.md]: changed to "config"

File: snap/README.md (hunk 2 above)
Comment (siggiskulason) [reply snap/README.md]: Good point - I added some more information..

**Review by tonyespy — APPROVED** [review]
LGTM

**Pull request merged**